force-app/main/default/classes/CON_DeleteContactOverride_CTRL.cls

```diff
@@ -315,21 +371,10 @@ public with sharing class CON_DeleteContactOverride_CTRL {
         public ContactCascadeDelete(Contact contactToDelete) {
             this.contactToDelete = contactToDelete;
 
-            relatedCases = [
-                SELECT CaseNumber, ContactId
-                FROM Case
-                WHERE ContactId = :contactToDelete.Id
-            ];
-            relatedOpportunities = [
-                SELECT Name, AccountId, Primary_Contact__c, Primary_Contact__r.AccountId, IsWon, IsClosed
-                FROM Opportunity
-                WHERE Primary_Contact__c = :contactToDelete.Id
-            ];
-            relatedRecurringDonations = [
-                SELECT Name, npe03__Contact__c
-                FROM npe03__Recurring_Donation__c
-                WHERE npe03__Contact__c = :contactToDelete.Id
-            ];
+            CON_DeleteContactOverrideSelector contactOverrideSelector = new CON_DeleteContactOverrideSelector();
```

Comment:
Instead of new'ing up additional instances of the selector class in these inner classes, the instance in the parent controller could be passed in as a second param to the constructor.  

`new ContactCascadeDelete(contactRecord, contactOverrideSelector);`